OPAL/si/src/main/scala/org/opalj/fpcf/scheduling/IndependentPhaseMergeScheduling.scala

```diff
@@ -0,0 +1,306 @@
+package org.opalj.fpcf.scheduling
+
+import org.opalj.collection.IntIterator
+import org.opalj.fpcf.AnalysisScenario
+import org.opalj.fpcf.ComputationSpecification
+import org.opalj.fpcf.LazyComputation
+import org.opalj.fpcf.PhaseConfiguration
+import org.opalj.fpcf.PropertyBounds
+import org.opalj.fpcf.PropertyStore
+import org.opalj.fpcf.Transformer
+import org.opalj.graphs.sccs
+
+/**
+ * Single Phase Scheduling (SPS) Strategy.
+ * Schedules all computations in a single batch without considering dependencies.
+ */
+class IndependentPhaseMergeScheduling[A](ps: PropertyStore, scheduleLazyTransformerInAllBatches: Boolean)
+    extends AnalysisScenario[A](ps) with SchedulingStrategy[A] {
+
+    val name = "IPMS"
+
+    override def schedule(
+        ps:    PropertyStore,
+        allCS: Set[ComputationSpecification[A]]
+    ): List[PhaseConfiguration[A]] = {
+
+        val lazyType = LazyComputation
+        val transformerType = Transformer
+
+        // Creates a map from ComputationSpecifications to their indices
+        val computationSpecificationMap: Map[ComputationSpecification[A], Int] = allCS.zipWithIndex.toMap
+        val computationSpecificationArray: Array[ComputationSpecification[A]] = allCS.toArray
+
+        // Initializes an empty schedule graph
+        var scheduleGraph: Map[Int, Set[Int]] = Map.empty
+
+        // Helper functions and logic for IPMS scheduling follow...
+        def getAllCSFromPropertyBounds(
```

Comment:
Extract common code / helper functions to `SchedulingStrategy` as much as possible.